### ISSUE TITLE
Doesn't work on Node.js

### ISSUE THREAD

**Issue by jcbhmr** [jimmywarting/await-sync#3]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `await-sync@1.0.1` for the project I'm working on.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/await-sync/mod.js b/node_modules/await-sync/mod.js
index 867ee90..0a31830 100644
--- a/node_modules/await-sync/mod.js
+++ b/node_modules/await-sync/mod.js
@@ -24,6 +24,7 @@ function createWorker (signal) {
     worker.postMessage({ port: remotePort, code: source, ab }, [remotePort])
 
     return function runSync (...args) {
+      Atomics.store(int32, 0, 0)
       // Send the arguments to the worker thread
       localPort.postMessage(args)
       // Wait for the worker thread to send the result back
diff --git a/node_modules/await-sync/worker.js b/node_modules/await-sync/worker.js
index fdb4f16..a507084 100644
--- a/node_modules/await-sync/worker.js
+++ b/node_modules/await-sync/worker.js
@@ -24,8 +24,7 @@ addEventListener('message', async evt => {
     }
   }
 
-  const blob = new Blob([code], { type: 'text/javascript' })
-  const url = URL.createObjectURL(blob)
+  const url = "data:text/javascript," + encodeURIComponent(code)
   const { default: fn } = await import(url)
 
   port.onmessage = async function onmessage (evt) {
@@ -50,3 +49,16 @@ addEventListener('message', async evt => {
     write2(u8)
   }
 })
+
+export {};
+if (typeof process !== "undefined") {
+  // You can't console.log() anything since all I/O goes THROUGH THE MAIN THREAD
+  // which is blocked by Atomics.wait(). Thus, we need to write somewhere else.
+  const { appendFileSync } = await import("node:fs");
+  process.on("uncaughtExceptionMonitor", e => {
+    appendFileSync("error.log", e.stack + "\n");
+  });
+  process.on("unhandledRejection", e => {
+    appendFileSync("error.log", e.stack + "\n");
+  });
+}
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by jimmywarting** [comment]
thx. something must have change with NodeJS core cuz it used to work before.

**Comment by jimmywarting** [comment]
released and fixed in 1.0.2
